patternlab/styleguide/source/_patterns/03-organisms/by-template/key-message.twig

```diff
@@ -1,46 +1,116 @@
 
-<section id="{{keyMessage.id}}"  class="ma__key-message {{ textColor ? 'ma__key-message--light-text' : '' }}{% if bgColor %} ma__key-message--{{ bgColor }}{% endif %}{% if bgImage %} ma__key-message--image{% endif %}">
+{% if ( theme == "c-primary" ) or ( theme == "c-primary-alt" ) %}
+  {% set buttontheme = { theme: "c-white" } %}
+{% elseif ( theme == "c-primary-alt-lightest" ) %}
+  {% set buttontheme = { theme: "c-primary-alt" } %}
+{% elseif ( theme == "c-white" ) or ( theme == "c-gray-lightest" ) %}
+  {% set buttontheme = { theme: "c-primary" } %}
+{% else %}
+  {% set buttontheme = { theme: theme } %}
+{% endif %}
 
-  {% if keyMessage.bgImage %}
+<section id="{{ id }}"  class="ma__key-message{{ msgClass }}{% if bgImage %} ma__key-message--image{% endif %}{% if keyMessage.textOverlay %} ma__key-message--image-overlay{% endif %}{% if positionTop %} ma__key-message--top{% endif %}">
+  {% if bgImage and positionTop %}
+    <div id="img{{ id }}"  class="ma__key-message__inline-image">
+    </div>
     <style>
-      #{{ keyMessage.id }}.ma__key-message--image {
-        background-image: url('{{ keyMessage.bgImage }}');
+      @media(min-width: 781px) {
+        #{{ id }}.ma__key-message--image {
+          background-image: url('{{ bgImage }}');
+        }
       }
-
-      {% if keyMessage.mobileBgImage %}
-      @media (-webkit-min-device-pixel-ratio: 2) {
-        #{{ keyMessage.id }}.ma__key-message--image {
-          background-image: url('{{ keyMessage.mobileBgImage }}');
+    </style>
+  {% elseif bgImage %}
+    <style>
+      #{{ id }}.ma__key-message--image {
+          background-image: url('{{ bgImage }}');
         }
       }
-      {% endif %}
     </style>
   {% endif %}
-
-  <div class="ma__key-message--wrapper">
-    {% set headingLevel = keyMessage.positionTop ? 1 : keyMessage.compHeading.level %}
-    <h{{ headingLevel }} class="ma__key-message--title">{{ keyMessage.compHeading.title }}</h{{ headingLevel }}>
-
-    {% if keyMessage.subtitle %}
-    <span class="ma__key-message--sub-title">{{ keyMessage.subtitle }}</span>
-    {% endif %}
-
-    {% if keyMessage.description %}
-    <div class="ma__key-message--description">
-      {% set richText = keyMessage.description %}
-      {% include "@organisms/by-author/rich-text.twig" %}    
-    </div>
-    {% endif %}
-
-    {% if keyMessage.button.text %}
-    <div class="ma__key-message--cta">
-      {% set button = keyMessage.button %}
-      {% include "@atoms/01-buttons/button.twig" %}
+  {% if keyMessage.mobileBgImage %}
+    <style>
+      #img{{ id }}.ma__key-message__inline-image {
+        background-image: url('{{ keyMessage.mobileBgImage }}');
+      }
+    </style>
+  {% elseif bgImage %}
+    <style>
+      #img{{ id }}.ma__key-message__inline-image {
+        background-image: url('{{ bgImage }}');
+      }
+    </style>
+  {% endif %}
+  {% if bgImage and positionTop %}
```

Comment:
Note: This is not ideal but the only way to set a margin based on the variable height of the overlay box's content.